### PR TITLE
Setting positions of objects works now. Also objects (cars and obstacles) get sent dynamically

### PR DESCRIPTION
**Pull request by BeeHiveJava**
The ball still needs to be added.

### CODE REVIEW

File: Soccar [Game-Server]/src/nl/soccar/gameserver/controller/socnet/message/SpawnObstacleMessage.java [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+
+    private final float x;
+    private final float y;
+    private final float angle;
+
+    private final float width;
+    private final float height;
+    private final ObstacleType type;
+
+    public SpawnObstacleMessage(float x, float y, float angle, float width, float height, ObstacleType type) {
```
Comment (ghost) [inline Soccar [Game-Server]/src/nl/soccar/gameserver/controller/socnet/message/SpawnObstacleMessage.java:23]: MINOR Document this public constructor. [rule](http://194.135.91.67:9000/coding_rules#rule_key=squid%3AUndocumentedApi)

File: Soccar [Game-Server]/src/nl/soccar/gameserver/util/CarUtilities.java [hunk 2]
```diff
@@ -18,23 +19,26 @@
     private CarUtilities() {
     }
 
-    public static void addCars(GameEngine engine, Map map, Team blue, Team red) {
-        addCars(engine, map, blue);
-        addCars(engine, map, red);
+    public static void addCars(GameWrapper game, Team blue, Team red) {
```
Comment (ghost) [inline Soccar [Game-Server]/src/nl/soccar/gameserver/util/CarUtilities.java:22]: MINOR Document this public method. [rule](http://194.135.91.67:9000/coding_rules#rule_key=squid%3AUndocumentedApi)

File: Soccar [Game-Server]/src/nl/soccar/gameserver/util/MapUtilities.java [hunk 3]
```diff
@@ -27,7 +27,7 @@ public static void addWalls(GameEngine engine, Map map) {
         addWestWalls(engine, map, mapHeight);
         addEastWalls(engine, map, mapWidth, mapHeight);
         addNorthAndSouthWalls(engine, mapWidth, mapHeight);
-        addCornerWalls(engine, mapWidth, mapHeight);
+        //addCornerWalls(engine, mapWidth, mapHeight);
```
Comment (ghost) [inline Soccar [Game-Server]/src/nl/soccar/gameserver/util/MapUtilities.java:30]: MAJOR This block of commented-out lines of code should be removed. [rule](http://194.135.91.67:9000/coding_rules#rule_key=squid%3ACommentedOutCodeLine)

File: Soccar [Game-Server]/src/nl/soccar/gameserver/controller/socnet/message/SpawnCarMessage.java [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+ */
+@MessageEvent(id = MessageConstants.SPAWN_CAR_MESSAGE_ID, handler = SpawnCarMessageHandler.class)
+public final class SpawnCarMessage extends Message {
+
+    private final int playerId;
+    private final float x;
+    private final float y;
+    private final float angle;
+
+    public SpawnCarMessage(int playerId, float x, float y, float angle) {
```
Comment (ghost) [inline Soccar [Game-Server]/src/nl/soccar/gameserver/controller/socnet/message/SpawnCarMessage.java:19]: MINOR Document this public constructor. [rule](http://194.135.91.67:9000/coding_rules#rule_key=squid%3AUndocumentedApi)

File: Soccar [Game-Server]/src/nl/soccar/gameserver/controller/socnet/message/PlayerChangedGameStatusMessage.java [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+
+/**
+ * @author PTS34A
+ */
+@MessageEvent(id = MessageConstants.PLAYER_CHANGED_GAME_STATUS_MESSAGE_ID, handler = PlayerChangedGameStatusMessageHandler.class)
+public final class PlayerChangedGameStatusMessage extends Message {
+
+    private final Status status;
+
+    public PlayerChangedGameStatusMessage(Status status) {
```
Comment (ghost) [inline Soccar [Game-Server]/src/nl/soccar/gameserver/controller/socnet/message/PlayerChangedGameStatusMessage.java:16]: MINOR Document this public constructor. [rule](http://194.135.91.67:9000/coding_rules#rule_key=squid%3AUndocumentedApi)

File: Soccar [Game-Server]/src/nl/soccar/gameserver/controller/socnet/message/PlayerChangedGameStatusMessage.java [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+    public Status getStatus() {
+        return status;
+    }
+
+    @Override
+    public int getId() {
+        return MessageConstants.PLAYER_CHANGED_GAME_STATUS_MESSAGE_ID;
+    }
+
+    public enum Status {
```
Comment (ghost) [inline Soccar [Game-Server]/src/nl/soccar/gameserver/controller/socnet/message/PlayerChangedGameStatusMessage.java:29]: MINOR Document this public enum. [rule](http://194.135.91.67:9000/coding_rules#rule_key=squid%3AUndocumentedApi)

File: Soccar [Game-Server]/src/nl/soccar/gameserver/model/session/GameWrapper.java [hunk 7]
```diff
@@ -18,39 +15,46 @@
...
     public GameWrapper(SessionWrapper session, Game game) {
         this.session = session;
         this.game = game;
 
         engine = new GameEngine(game);
-        timer = new Timer();
     }
 
-    public void start() {
+    public void requestStart() {
```
Comment (ghost) [inline Soccar [Game-Server]/src/nl/soccar/gameserver/model/session/GameWrapper.java:48]: MINOR Document this public method. [rule](http://194.135.91.67:9000/coding_rules#rule_key=squid%3AUndocumentedApi)

File: Soccar [Game-Server]/src/nl/soccar/gameserver/model/session/GameWrapper.java [hunk 8]
```diff
@@ -70,8 +74,16 @@ public void run() {
     public void stop() {
         timer.cancel();
 
-        game.stop();
         engine.stop();
+
+        playersReady.clear();
+    }
+
+    public void sendWorldObjects(PlayerWrapper player) {
```
Comment (ghost) [inline Soccar [Game-Server]/src/nl/soccar/gameserver/model/session/GameWrapper.java:82]: MINOR Document this public method. [rule](http://194.135.91.67:9000/coding_rules#rule_key=squid%3AUndocumentedApi)

File: Soccar [Game-Server]/src/nl/soccar/gameserver/model/session/GameWrapper.java [hunk 9]
```diff
@@ -18,39 +15,46 @@
...
+    private final List<PlayerWrapper> playersReady = new ArrayList<>();
+
     private final SessionWrapper session;
     private final Game game;
 
     private final GameEngine engine;
-    private final Timer timer;
+    private Timer timer;
 
     public GameWrapper(SessionWrapper session, Game game) {
```
Comment (ghost) [inline Soccar [Game-Server]/src/nl/soccar/gameserver/model/session/GameWrapper.java:41]: MINOR Document this public constructor. [rule](http://194.135.91.67:9000/coding_rules#rule_key=squid%3AUndocumentedApi)

File: Soccar [Game-Server]/src/nl/soccar/gameserver/model/session/GameWrapper.java [hunk 10]
```diff
@@ -70,8 +74,16 @@ public void run() {
...
+    }
+
+    public void sendWorldObjects(PlayerWrapper player) {
+        sendCars(player);
+        sendObstacles(player);
+
+        player.getConnection().send(new ChangePlayerStatusMessage(ChangePlayerStatusMessage.Status.READY_TO_PLAY));
     }
 
     public void movePlayer(PlayerWrapper player, SteerAction steerAction, HandbrakeAction handbrakeAction, ThrottleAction throttleAction) {
```
Comment (ghost) [inline Soccar [Game-Server]/src/nl/soccar/gameserver/model/session/GameWrapper.java:89]: MINOR Document this public method. [rule](http://194.135.91.67:9000/coding_rules#rule_key=squid%3AUndocumentedApi)

File: Soccar [Game-Server]/src/nl/soccar/gameserver/model/session/GameWrapper.java [hunk 11]
```diff
@@ -70,8 +74,16 @@ public void run() {
     public void stop() {
```
Comment (ghost) [inline Soccar [Game-Server]/src/nl/soccar/gameserver/model/session/GameWrapper.java:74]: MINOR Document this public method. [rule](http://194.135.91.67:9000/coding_rules#rule_key=squid%3AUndocumentedApi)

File: Soccar [Game-Server]/src/nl/soccar/gameserver/controller/socnet/message/handler/ChangePlayerStatusMessageHandler.java [hunk 12]
```diff
@@ -0,0 +1,55 @@
...
+        }
+
+        SessionWrapper session = player.getCurrentSession();
+        if (session == null) {
+            return;
+        }
+
+        ChangePlayerStatusMessage.Status status = message.getStatus();
+
+        System.out.println(status);
```
Comment (ghost) [inline Soccar [Game-Server]/src/nl/soccar/gameserver/controller/socnet/message/handler/ChangePlayerStatusMessageHandler.java:31]: MAJOR Replace this usage of System.out or System.err by a logger. [rule](http://194.135.91.67:9000/coding_rules#rule_key=squid%3AS106)

File: Soccar [Game-Server]/src/nl/soccar/gameserver/controller/socnet/message/ChangePlayerStatusMessage.java [hunk 13]
```diff
@@ -0,0 +1,34 @@
...
+
+/**
+ * @author PTS34A
+ */
+@MessageEvent(id = MessageConstants.CHANGE_PLAYER_STATUS_MESSAGE_ID, handler = ChangePlayerStatusMessageHandler.class)
+public final class ChangePlayerStatusMessage extends Message {
+
+    private final Status status;
+
+    public ChangePlayerStatusMessage(Status status) {
```
Comment (ghost) [inline Soccar [Game-Server]/src/nl/soccar/gameserver/controller/socnet/message/ChangePlayerStatusMessage.java:16]: MINOR Document this public constructor. [rule](http://194.135.91.67:9000/coding_rules#rule_key=squid%3AUndocumentedApi)

File: Soccar [Game-Server]/src/nl/soccar/gameserver/controller/socnet/message/ChangePlayerStatusMessage.java [hunk 14]
```diff
@@ -0,0 +1,34 @@
...
+    public Status getStatus() {
+        return status;
+    }
+
+    @Override
+    public int getId() {
+        return MessageConstants.CHANGE_PLAYER_STATUS_MESSAGE_ID;
+    }
+
+    public enum Status {
```
Comment (ghost) [inline Soccar [Game-Server]/src/nl/soccar/gameserver/controller/socnet/message/ChangePlayerStatusMessage.java:29]: MINOR Document this public enum. [rule](http://194.135.91.67:9000/coding_rules#rule_key=squid%3AUndocumentedApi)

**Comment by ghost** [comment]
SonarQube analysis reported 27 issues
* MAJOR 3 major
* MINOR 24 minor

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. MAJOR [MapUtilities.java#L129](https://github.com/PTS3-S34A/Game-Server/blob/67945a7e06e0b15072f8fa0081de98970457aabd/Soccar [Game-Server]/src/nl/soccar/gameserver/util/MapUtilities.java#L129): Remove this unused private "addCornerWalls" method. [rule](http://194.135.91.67:9000/coding_rules#rule_key=squid%3AUnusedPrivateMethod)
1. MINOR [GameServerSocnetController.java#L17](https://github.com/PTS3-S34A/Game-Server/blob/67945a7e06e0b15072f8fa0081de98970457aabd/Soccar [Game-Server]/src/nl/soccar/gameserver/controller/socnet/GameServerSocnetController.java#L17): Document this public constructor. [rule](http://194.135.91.67:9000/coding_rules#rule_key=squid%3AUndocumentedApi)
1. MINOR [GameServerSocnetController.java#L21](https://github.com/PTS3-S34A/Game-Server/blob/67945a7e06e0b15072f8fa0081de98970457aabd/Soccar [Game-Server]/src/nl/soccar/gameserver/controller/socnet/GameServerSocnetController.java#L21): Document this public method. [rule](http://194.135.91.67:9000/coding_rules#rule_key=squid%3AUndocumentedApi)
1. MINOR [GameServerSocnetController.java#L50](https://github.com/PTS3-S34A/Game-Server/blob/67945a7e06e0b15072f8fa0081de98970457aabd/Soccar [Game-Server]/src/nl/soccar/gameserver/controller/socnet/GameServerSocnetController.java#L50): Document this public method. [rule](http://194.135.91.67:9000/coding_rules#rule_key=squid%3AUndocumentedApi)
1. MINOR [GameServerSocnetController.java#L54](https://github.com/PTS3-S34A/Game-Server/blob/67945a7e06e0b15072f8fa0081de98970457aabd/Soccar [Game-Server]/src/nl/soccar/gameserver/controller/socnet/GameServerSocnetController.java#L54): Document this public method. [rule](http://194.135.91.67:9000/coding_rules#rule_key=squid%3AUndocumentedApi)
1. MINOR [RoomWrapper.java#L37](https://github.com/PTS3-S34A/Game-Server/blob/67945a7e06e0b15072f8fa0081de98970457aabd/Soccar [Game-Server]/src/nl/soccar/gameserver/model/session/RoomWrapper.java#L37): Document this public constructor. [rule](http://194.135.91.67:9000/coding_rules#rule_key=squid%3AUndocumentedApi)
1. MINOR [RoomWrapper.java#L42](https://github.com/PTS3-S34A/Game-Server/blob/67945a7e06e0b15072f8fa0081de98970457aabd/Soccar [Game-Server]/src/nl/soccar/gameserver/model/session/RoomWrapper.java#L42): Document this public method. [rule](http://194.135.91.67:9000/coding_rules#rule_key=squid%3AUndocumentedApi)
1. MINOR [RoomWrapper.java#L79](https://github.com/PTS3-S34A/Game-Server/blob/67945a7e06e0b15072f8fa0081de98970457aabd/Soccar [Game-Server]/src/nl/soccar/gameserver/model/session/RoomWrapper.java#L79): Document this public method. [rule](http://194.135.91.67:9000/coding_rules#rule_key=squid%3AUndocumentedApi)
1. MINOR [RoomWrapper.java#L113](https://github.com/PTS3-S34A/Game-Server/blob/67945a7e06e0b15072f8fa0081de98970457aabd/Soccar [Game-Server]/src/nl/soccar/gameserver/model/session/RoomWrapper.java#L113): Document this public method. [rule](http://194.135.91.67:9000/coding_rules#rule_key=squid%3AUndocumentedApi)
1. MINOR [RoomWrapper.java#L131](https://github.com/PTS3-S34A/Game-Server/blob/67945a7e06e0b15072f8fa0081de98970457aabd/Soccar [Game-Server]/src/nl/soccar/gameserver/model/session/RoomWrapper.java#L131): Document this public method. [rule](http://194.135.91.67:9000/coding_rules#rule_key=squid%3AUndocumentedApi)

File: Soccar [Game-Server]/src/nl/soccar/gameserver/util/MapUtilities.java (hunk 3 above)
Comment (Laugslander) [reply Soccar [Game-Server]/src/nl/soccar/gameserver/util/MapUtilities.java:30]: addCornerWalls methode verwijderen?

**Review by Laugslander — CHANGES_REQUESTED** [review]

**Review by Laugslander — APPROVED** [review]

**Pull request merged**